charts/temporal/templates/_helpers.tpl

```diff
@@ -42,7 +42,7 @@ Create the name of the service account
 Define the service account as needed
 */}}
 {{- define "temporal.serviceAccount" -}}
-{{- if .Values.serviceAccount.name -}}
+{{- if .Values.serviceAccount.create -}}
```

Comment:
Makes sense, but we need to still use the service account if `create` is false but `name` is set.

Reply:
This prevents users from using a pre-made service account. Helm will fail trying to create the serviceAccount that already exists.

Reply:
Sorry, I don't understand your fix. Now we might create a serviceAccount called "default". I don't understand the reasoning there. Please see if https://github.com/temporalio/helm-charts/pull/574 fixes the behaviour you needed.